src/todos/todos.repository.ts

```diff
@@ -0,0 +1,22 @@
+import { Model } from 'mongoose';
+import { Inject, Injectable } from '@nestjs/common';
+import { Todo } from './interfaces/todo.interface';
+import { CreateTodoDto } from './dtos/request/create-todo.dto';
+
+@Injectable()
+export class TodosRepository {
```

Comment:
I have created a new layer to separate responsibilities. 

The controler handels the request, 

The service handles the business logic(the creation of the data structure for the consumer[front end]),

The repository handles the data base connection.